src/Android/Avalonia.Android/AvaloniaAndroidApplication.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Runtime;
+
+namespace Avalonia.Android
+{
+    internal interface IAndroidApplication
+    {
+        SingleViewLifetime? Lifetime { get; set; }
+    }
+
+    public class AvaloniaAndroidApplication<TApp> : global::Android.App.Application, IAndroidApplication
+        where TApp : Application, new()
+    {
+        SingleViewLifetime? IAndroidApplication.Lifetime { get; set; }
+
+        protected AvaloniaAndroidApplication(nint javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
+        {
+        }
+
+        public override void OnCreate()
+        {
+            base.OnCreate();
+            InitializeAppLifetime();
+        }
+
+        private void InitializeAppLifetime()
+        {
+            var builder = CreateAppBuilder();
+            builder = CustomizeAppBuilder(builder);
+
+            var lifetime = new SingleViewLifetime();
+
+            (this as IAndroidApplication)!.Lifetime = lifetime;
```

Comment:
Use `((IAndroidApplication)this)` instead of `as` + `!`.

Reply:
Updated.